src/main/java/ekud/task/TaskList.java

```diff
@@ -0,0 +1,154 @@
+package ekud.task;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Vector;
+
+import ekud.common.exception.InvalidTaskIndexException;
+
+/**
+ * Iterable for containing any task type
+ */
+public class TaskList implements Iterable<Task> {
+    private final Vector<Task> taskArray;
+    private int doneCount;
+
+    /**
+     * Construct a new TaskList from existing tasks.
+     *
+     * @param taskArray The array containing existing tasks.
+     */
+    public TaskList(Collection<Task> taskArray) {
+        this.taskArray = new Vector<>(taskArray);
+        for (Task task : taskArray) {
+            if (task.isDone) {
+                ++doneCount;
+            }
+        }
+    }
+
+    /**
+     * Construct a new TaskList with no existing tasks.
+     */
+    public TaskList() {
+        taskArray = new Vector<>();
+        doneCount = 0;
+    }
+
+    /**
+     * Add a new task to the end of the existing list.
+     *
+     * @param newTask The task to be added.
+     */
+    public void add(Task newTask) {
+        taskArray.add(newTask);
+    }
+
+    /**
+     * Remove a task by index.
+     *
+     * @param index The index of the task to be removed.
+     * @return Task removed from list.
+     * @throws InvalidTaskIndexException If the index is out of range (index < 0 || index >= size()).
```

Comment:
I like that the error thrown is very self explanatory!

Reply:
Agreed with this reviewer, it's easy to understand what errors are being expected especially with your javadoc comment too